### ISSUE TITLE
[terra-application-links] More button isn't understood by assistive technology

### ISSUE THREAD

**Issue by neilpfeiffer** [cerner/terra-framework#1579]
# Feature Request

## Description

In current component `terra-application-links` used by the older/previous navigation component `terra-application-layout`, when more tabs appear than can fit visibly, Terra creates a "More" menu. The current "More" menu button is not properly understood by by assistive technology in a way that communicates its correction interactions. 

### How to Reproduce
- While Voiceover is enabled, tab to the More tab
  - Assistive technology will read, "More, selected, tab, group, 1 of 1".
- Press Return or Space when focused on the More tab, and try tabbing through the menu
  - You will be unable to access the menu or any other focusable elements until pressing Esc

### Proposal

- Change role from being read as tab+group to a button that opens a menu
- Remove that it reports as being "selected"
- Make menu items accessible for keyboard users

## Additional Context / Screenshots

**Documented criteria:**
HNCONSUMER-2599

**Full Tech Design of related issues compiled here:**
[consumer-framework-tech-designs / spikes / HNCONSUMER-28288: Investigate accessibility issues with navigation.md](https://github.cerner.com/healtheintent/consumer-framework-tech-designs/blob/c709a5bba99a97b0b6f39805e3d2db0fae2f82c7/spikes/HNCONSUMER-28288:%20Investigate%20accessibility%20issues%20with%20navigation.md#more-button-isnt-understood-by-assistive-technology)

## @ Mentions

Will be contribution from Consumer development team
